src/node_file.cc

```diff
@@ -1942,7 +1942,6 @@ static inline Maybe<void> CheckOpenPermissions(Environment* env,
   const int write_as_side_effect = flags & (UV_FS_O_APPEND | UV_FS_O_CREAT |
                                             UV_FS_O_TRUNC | UV_FS_O_TEMPORARY);
 
-  // TODO(rafaelgss): it can be optimized to avoid two permission checks
```

Comment:
I didn't understand why this todo doesn't apply anymore. Can you explain it? It seems we have 2 read and write checks and looks like it can be merged?

Reply:
If we merge it, it won't be accurate since we'll need to check the `rwflags` anyway.

I've added this TODO before https://github.com/nodejs/node/commit/aa30e16716ee9caa6503c7a7bf2e27aa66ce5120 change.

We would need to iterate over 2 trees anyway.